com.unity.netcode.gameobjects/Runtime/SceneManagement/SceneEventData.cs

```diff
@@ -323,6 +341,32 @@ private int SortNetworkObjects(NetworkObject first, NetworkObject second)
             return 0;
         }
 
+        /// <summary>
+        /// Sorts the synchronization order of the NetworkObjects to be serialized
+        /// by parents before children.
+        /// </summary>
+        /// <remarks>
+        /// This only handles late joining players. Spawning and nesting several children
+        /// dynamically is still handled by the orphaned child list when deserialized out of
+        /// hierarchical order (i.e. Spawn parent and child dynamically, parent message is
+        /// dropped and re-sent but child object is received and processed)
+        /// </remarks>
+        private int SortParentedNetworkObjects(NetworkObject first, NetworkObject second)
+        {
+            // If the first has a parent, move the first down
+            if (first.transform.parent != null)
```

Comment:
if they BOTH have parents, return 0? This logic will currently return 1.

Reply:
The sorting occurs on the first object.
If the first object has a parent, then it is a child and move it down.
Otherwise...if the first does not have a parent but the second does have a parent then move the first up (i.e. children move down the list).
If neither the first or the second have a parent then don't move the first element.